src/api/ums/umsModule.ts

```diff
@@ -0,0 +1,81 @@
+import { ReflectiveInjector } from "injection-js";
+import { API } from "../../config";
+import { Methods, RequestAdapter } from "../../internal/requestAdapter";
+import { Client, ClientInit } from "../core/client";
+import { IModule } from "../core/module";
+import { AuthOptions, TokenManager } from "../core/tokenManager";
+
+export interface IUMSSignInOptions {
+  email: string;
+  password: string;
+  default?: boolean;
+  auth?: string;
+}
+
+export type IUMSCredentials = Pick<IUMSSignInOptions, 'email' | 'password'>;
+
+export interface IUMSUser {
+  id: string;
+  email: string;
+}
+
+export class UMSModule implements IModule {
+
+  private tokenManager: TokenManager;
+
+  private requestAdapter: RequestAdapter;
+
+  private systemInit: Promise<Client>;
+
+  private projectId: string;
+
+  public init(coreInjector: ReflectiveInjector) {
+    this.tokenManager = coreInjector.get(TokenManager);
+    this.requestAdapter = coreInjector.get(RequestAdapter);
+    this.systemInit = coreInjector.get(ClientInit);
+    this.projectId = coreInjector.get(AuthOptions).projectID;
+
+    return Promise.resolve(this);
+  }
+
+  public terminate() {
```

Comment:
same here

Reply:
the same, this method is described in the interface that we implemented